implementation/src/main/java/io/smallrye/metrics/setup/config/MetricsConfigurationManager.java

```diff
@@ -0,0 +1,320 @@
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{0}\" with values: {1} ",
+                        new Object[] { MP_HISTOGRAM_MIN_CONFIG, metricName, retVal });
+            }
+            return retVal;
+        }
+        return null;
+
+    }
+
+    /**
+     * Returns the matching configuration object if it exists, null otherwise
+     * 
+     * @param metricName the metric name to check configuration against
+     * @return the matching configuration object if it exists, null otherwise
+     */
+    public synchronized TimerBucketMaxConfiguration getDefaultTimerMaxBucketConfiguration(String metricName) {
+
+        String appName = getApplicationName();
+
+        Collection<TimerBucketMaxConfiguration> computedValues = defaultTimerBucketMaxConfig.computeIfAbsent(appName,
+                f -> {
+                    Optional<String> input = ConfigProvider.getConfig().getOptionalValue(MP_TIMER_MAX_CONFIG, String.class);
+
+                    return (input.isPresent()) ? TimerBucketMaxConfiguration.parse(input.get()) : null;
+
+                });
+
+        if (computedValues != null && computedValues.size() != 0) {
+            TimerBucketMaxConfiguration retVal = TimerBucketMaxConfiguration.matches(computedValues, metricName);
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{0}\" with values: {1} ",
+                        new Object[] { MP_TIMER_MAX_CONFIG, metricName, retVal });
+            }
+            return retVal;
+        }
+        return null;
+
+    }
+
+    /**
+     * Returns the matching configuration object if it exists, null otherwise
+     * 
+     * @param metricName the metric name to check configuration against
+     * @return the matching configuration object if it exists, null otherwise
+     */
+    public synchronized TimerBucketMinConfiguration getDefaultTimerMinBucketConfiguration(String metricName) {
+
+        String appName = getApplicationName();
+
+        Collection<TimerBucketMinConfiguration> computedValues = defaultTimerBucketMinConfig.computeIfAbsent(appName,
+                f -> {
+                    Optional<String> input = ConfigProvider.getConfig().getOptionalValue(MP_TIMER_MIN_CONFIG, String.class);
+
+                    return (input.isPresent()) ? TimerBucketMinConfiguration.parse(input.get()) : null;
+
+                });
+
+        if (computedValues != null && computedValues.size() != 0) {
+            TimerBucketMinConfiguration retVal = TimerBucketMinConfiguration.matches(computedValues, metricName);
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{0}\" with values: {1} ",
```

Comment:
Should the parameter be {0}, {1}, and {2}?

Reply:
fixed